services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -2287,6 +2278,16 @@ private Pair<Boolean, Version> addVersion(
           }
         }
       }
+
+      if (startIngestion) {
+        // Store write lock is released before polling status
+        waitUntilNodesAreAssignedForResource(
```

Comment:
Since cluster-level lock is released, controller leader->standby transition thread could interrupt, acquire the cluster-level lock and set clusterResources to null. Then `waitUntilNodesAreAssignedForResource` will throw exception during `getHelixVeniceClusterResources` and addVersion will fail.

Since addVersion is not idempotent and version number is already bumped, controller won't retry addVersion. We would observe more addVersion failure during controller leadership handover.

Reply:
The problem that Shuhui mentioned makes sense to me. It brings a question in my mind on the related topic, if a controller is transferring from leader -> standby, but couldn't finish for quite some time, because it needs to wait for the cluster writer lock to release, while addVersion is holding it, then would it also be a concern that we could have a controller split brain issue for quite long time, since a new controller leader is already in charge?

Reply:
Talked with Shuhui offline; added the check for clusterResource being null and not fail the addVersion command in this case; instead, new leader controller will start monitoring the push.

Reply:
Talked with Lei offline; ideally, the old leader will step down before a new leader is selected.